### PR TITLE
Initial version of GitPacker using config provided via json file.

### PR DESCRIPTION
**Pull request by galxy25**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+		// Exit early if any errors cloning
+		return
+	}
+	// If specified, compress all cloned repositories
+	// into a single zip file
+	if !packConfig.Archive {
+		return
+	}
+	if packConfig.ArchiveFilename == "" {
+		fmt.Println("Must specific archive_filename if archive is true.")
```
Comment (efabens) [inline main.go:122]: /s/specific/specify <- I think 🤷

File: main.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+		if info.IsDir() {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		// Ensure that `path` is not absolute; it should not start with "/".
```
Comment (efabens) [inline main.go:176]: What happens if path is absolute and should we do a starts with check to verify that it doesn't being with `/`

**Review by efabens — APPROVED** [review]

File: main.go (hunk 1 above)
Comment (galxy25) [reply main.go:122]: good catch

File: main.go (hunk 2 above)
Comment (galxy25) [reply main.go]: Updated to validate the path we're given doesn't begin with /.

**Review by lkwdwrd — APPROVED** [review]
Approved pending we make an update to either shallow clone the repositories or remove the `.git/` directory before zipping. Both are good -- shallow clone will run faster (sometimes a _lot_ faster if it is a big repo), and removing the .git directory will result in smaller artifacts at the end of the day. Preference for shallow clone as it also helps with artifact size.

**Pull request merged**